superset-frontend/src/explore/controlUtils.js

```diff
@@ -52,21 +52,18 @@ export function validateControl(control, processedState) {
  * Find control item from control panel config.
  */
 function findControlItem(controlPanelSections, controlKey) {
-  for (const section of controlPanelSections) {
-    for (const controlArr of section.controlSetRows) {
-      for (const control of controlArr) {
-        if (controlKey === control) return control;
-        if (
-          control !== null &&
-          typeof control === 'object' &&
-          control.name === controlKey
-        ) {
-          return control;
-        }
-      }
-    }
-  }
-  return null;
+  return (
```

Comment:
This one is much more opaque to me and this seems more risky than the other pieces of logic in this PR.  I checked and there is not unit test on this method. Can we either 1) write a covering unit test that works before/after or 2) minimize the changes. 

It looks like there are existing fixtures for `controlPanelSections` that can be refactored out of superset-frontend/spec/javascripts/explore/controlUtils_spec.jsx .

Reply:
I added some tests and created a separate fixtures file for controlPanelSections

Reply:
They pass on both old and refactored version of this function